tracing-subscriber/src/fmt/writer.rs

```diff
@@ -103,6 +104,60 @@ impl MakeWriter for TestWriter {
     }
 }
 
+/// A writer that erases the specific [`io::Write`] and [`Makewriter`] types being used.
+///
+/// This is useful in cases where the concrete type of the writer cannot be known
+/// until runtime.
+///
+/// [`MakeWriter`]: trait.MakeWriter.html
+/// [`io::Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
+pub struct BoxMakeWriter {
+    inner: Box<dyn MakeWriter<Writer = Box<dyn Write>> + Send + Sync>,
+}
+
+impl BoxMakeWriter {
+    /// Accepts an instance of [`MakeWriter`] and returns a BoxMakeWriter that wraps it.
+    ///
+    /// [`MakeWriter`]: trait.MakeWriter.html
+    pub fn new<M>(make_writer: M) -> Self
+    where
+        M: MakeWriter + Send + Sync + 'static,
+        M::Writer: Write + 'static,
+    {
+        Self {
+            inner: Box::new(Boxed(make_writer)),
+        }
+    }
+}
+
+impl Debug for BoxMakeWriter {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "BoxMakeWriter")
```

Comment:
I'm not entirely sure what to have as the message here.

Reply:
This seems fine. I might write something like
```suggestion
        f.pad("BoxMakeWriter { ... }")
```
which is what the standard library [tends to do](https://doc.rust-lang.org/src/std/io/util.rs.html#127-131) for types without interesting `Debug` output — this just makes it a little clearer that it's a struct with some opaque contents.